sbt_common/storage/src/test/scala/uk/ac/wellcome/storage/vhs/StringStoreVersionedHybridStoreTest.scala

```diff
@@ -0,0 +1,164 @@
+package uk.ac.wellcome.storage.vhs
+
+import com.gu.scanamo.Scanamo
+import com.gu.scanamo.syntax._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{FunSpec, Matchers}
+import uk.ac.wellcome.storage.s3.S3StringStore
+import uk.ac.wellcome.storage.test.fixtures.LocalDynamoDb.Table
+import uk.ac.wellcome.storage.test.fixtures.LocalVersionedHybridStore
+import uk.ac.wellcome.storage.test.fixtures.S3.Bucket
+import uk.ac.wellcome.test.fixtures._
+import uk.ac.wellcome.test.utils.ExtendedPatience
+import uk.ac.wellcome.utils.GlobalExecutionContext._
+
+import scala.util.Random
+
+class StringStoreVersionedHybridStoreTest
+    extends FunSpec
+    with Matchers
+    with ScalaFutures
+    with ExtendedPatience
+    with LocalVersionedHybridStore {
+
+  import uk.ac.wellcome.storage.dynamo._
+
+  def withS3StringStoreFixtures[R](
+    testWith: TestWith[(Bucket,
+                        Table,
+                        VersionedHybridStore[String, S3StringStore]),
+                       R]
+  ): R =
+    withLocalS3Bucket[R] { bucket =>
+      withLocalDynamoDbTable[R] { table =>
+        withStringVHS[R](bucket, table) { vhs =>
+          testWith((bucket, table, vhs))
+        }
+      }
+    }
+
+  // The StringStore demonstrates the base functionality of the VHS
+
+  describe("with S3StringStore") {
+    it("stores a record if it has never been seen before") {
+      withS3StringStoreFixtures {
+        case (bucket, table, hybridStore) =>
+          val id = Random.nextString(5)
+          val record = "One ocelot in orange"
+
+          val future = hybridStore.updateRecord(id)(record)(identity)()
+
+          whenReady(future) { _ =>
+            getContentFor(bucket, table, id) shouldBe record
+          }
+      }
+    }
+
+    it("applies the given transformation to an existing record") {
+      withS3StringStoreFixtures {
+        case (bucket, table, hybridStore) =>
+          val id = Random.nextString(5)
+          val record = "One ocelot in orange"
+
+          val expectedRecord = "Two tapping turtles"
+
+          val t = (e: String) => expectedRecord
+
+          val future =
+            hybridStore
+              .updateRecord(id)(record)(identity)()
+              .flatMap(_ => hybridStore.updateRecord(id)(record)(t)())
+
+          whenReady(future) { _ =>
+            getContentFor(bucket, table, id) shouldBe expectedRecord
+          }
+      }
+    }
+
+    it("updates DynamoDB and S3 if it sees a new version of a record") {
```

Comment:
Isn't this the same test case as above? And I don't see any assertion about the Dynamo content

Reply:
yup, noted.